src/plugins/dashboard/public/services/dashboard_backup/dashboard_backup_service.ts

```diff
@@ -52,6 +55,21 @@ class DashboardSessionStorageService implements DashboardSessionStorageServiceTy
     }
   }
 
+  public getViewMode = (): ViewMode => {
+    return this.localStorage.get(DASHBOARD_VIEWMODE_LOCAL_KEY);
+  };
+
+  public storeViewMode = (viewMode: ViewMode) => {
+    try {
+      this.localStorage.set(DASHBOARD_VIEWMODE_LOCAL_KEY, viewMode);
+    } catch (e) {
+      this.notifications.toasts.addDanger({
+        title: panelStorageErrorStrings.getPanelsGetError(e.message),
+        'data-test-subj': 'dashboardPanelsGetFailure',
+      });
```

Comment:
**nit:** Hmmm... Is `"Error encountered while fetching unsaved changes"` the right error here? Should it have a unique failure message?

Reply:
Resolved in ac111e6c016